### PR TITLE
ARROW-16093: [Python] Address docstrings in Filesystems (Python Implementations)

### PR DESCRIPTION
**Pull request by AlenkaF**

### CODE REVIEW

File: python/pyarrow/_fs.pyx [hunk 1]
```diff
@@ -1027,6 +1027,23 @@ cdef class PyFileSystem(FileSystem):
     ----------
     handler : FileSystemHandler
         The handler object implementing custom filesystem behavior.
+
+    Examples
+    --------
+    Create an fsspec-based filesystem object for Google Cloud Storage:
```
Comment (pitrou) [inline python/pyarrow/_fs.pyx:1033]: Looks like this fails the CI documentation tests as `gcsfs` isn't installed.

**Review by pitrou — COMMENTED** [review]
Thanks for the improvement. Here is a suggestion.

File: python/pyarrow/_fs.pyx (hunk 1 above)
Comment (AlenkaF) [reply python/pyarrow/_fs.pyx:1033]: Awesome, Github filesystem will be a great example. Thanks!!
And sorry for asking for review before waiting for the CI to finish - was a bit too sure of myself =)

File: python/pyarrow/_fs.pyx [hunk 2]
```diff
@@ -1027,6 +1027,23 @@ cdef class PyFileSystem(FileSystem):
...
+
+    >>> from fsspec.implementations import github
+    >>> gfs = github.GithubFileSystem('apache', 'arrow')
+
+    Get a PyArrow FileSystem object:
+
+    >>> from pyarrow.fs import PyFileSystem, FSSpecHandler
+    >>> pa_fs = PyFileSystem(FSSpecHandler(gfs))
+
+    Use :func:`~pyarrow.fs.FileSystem` functionality `get_file_info()`:
```
Comment (jorisvandenbossche) [inline python/pyarrow/_fs.pyx:1043]:
```suggestion
    Use :func:`~pyarrow.fs.FileSystem` functionality ``get_file_info()``:
```

Comment (AlenkaF) [reply python/pyarrow/_fs.pyx:1043]: I can't seem to remember to use double `` 🤦‍♀️  Thanks!

**Comment by AlenkaF** [comment]
The CI fail doesn't seem related.

File: python/pyarrow/_fs.pyx [hunk 3]
```diff
@@ -1030,20 +1030,20 @@ cdef class PyFileSystem(FileSystem):
...
     >>> from pyarrow.fs import PyFileSystem, FSSpecHandler
-    >>> pa_fs = PyFileSystem(FSSpecHandler(fs))
+    >>> pa_fs = PyFileSystem(FSSpecHandler(gfs))
 
     Use :func:`~pyarrow.fs.FileSystem` functionality `get_file_info()`:
 
-    >>> pa_fs.get_file_info('gcp-public-data-sentinel-2/index.csv.gz')
-    <FileInfo for 'gcp-public-data-sentinel-2/index.csv.gz': type=FileType.File, size=2783655355>
+    >>> pa_fs.get_file_info('README.md')
+    <FileInfo for 'README.md': type=FileType.File, size=5302>
```
Comment (pitrou) [inline python/pyarrow/_fs.pyx:1046]: Note that if you want this test to always succeed, you should choose a specific changeset in `GithubFileSystem`.

Comment (AlenkaF) [reply python/pyarrow/_fs.pyx:1046]: Will add.

**Review by pitrou — APPROVED** [review]
+1, but let's wait for CI (the docstring might need wrapping at 80 chars?)

**Pull request merged**